web/src/copy_and_paste.js

```diff
@@ -341,7 +346,30 @@ export function paste_handler(event) {
     }
 
     if (clipboardData.getData) {
+        const $textarea = $(event.currentTarget);
+        const paste_text = clipboardData.getData("text");
+        // Trim the paste_text to handle sloppy copying
+        const trimmed_paste_text = paste_text.trim();
         const paste_html = clipboardData.getData("text/html");
+
+        // Match for anything that resembles a URI, something://something
+        const uri_regex = /^\w+:\/\/\S+$/;
+        const range = $textarea.range();
+
+        // Only try to generate formatted links when dealing with a URI
+        // and a range selection. Note that even clipboards with "text/html"
+        // have a "text" equivalent, so we need an if statement that checks
+        // for more than a value on `trimmed_paste_text`
+        if (uri_regex.test(trimmed_paste_text) && range.text) {
+            const link_text = range.text;
+            const uri = trimmed_paste_text;
+            const markdown = paste_handler_link_formatter(uri, link_text);
+            event.preventDefault();
```

Comment:
I know it doesn't matter from a logical standpoint, but it seems odd to do these two things in the middle of the block.